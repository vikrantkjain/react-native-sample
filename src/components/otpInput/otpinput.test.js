import {render, fireEvent, screen} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import OTPTextView from './otpFC';
describe('Test OTP input', () => {
  const onChangeTestMock = jest.fn();
  const treeJSON = renderer.create(
    <OTPTextView handleTextChange={onChangeTestMock} />,
  );
  render(<OTPTextView handleTextChange={onChangeTestMock} />);
  it('Render successfully', () => {
    expect(treeJSON).toBeDefined();
  });
  it('Render correctly', () => {
    const treeToJSON = treeJSON.toJSON();
    expect(treeToJSON).toMatchSnapshot();
  });
});
