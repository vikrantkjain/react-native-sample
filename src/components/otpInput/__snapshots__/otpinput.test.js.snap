// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test OTP input Render correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      },
      {},
    ]
  }
>
  <TextInput
    autoCorrect={false}
    autoFocus={false}
    keyboardType="numeric"
    maxLength={1}
    multiline={false}
    onChangeText={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    placeholder=""
    placeholderTextColor="#E4E4E4"
    style={
      [
        {
          "borderBottomWidth": 2,
          "borderColor": "#E9E9E9",
          "color": "#000000",
          "fontFamily": "AvenirNext-Medium",
          "fontSize": 22,
          "height": 70,
          "marginRight": 4,
          "textAlign": "center",
          "width": 60,
        },
        {},
        {
          "borderColor": "#333333",
        },
      ]
    }
    textAlign="center"
    value=""
  />
  <TextInput
    autoCorrect={false}
    autoFocus={false}
    keyboardType="numeric"
    maxLength={1}
    multiline={false}
    onChangeText={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    placeholder=""
    placeholderTextColor="#E4E4E4"
    style={
      [
        {
          "borderBottomWidth": 2,
          "borderColor": "#E9E9E9",
          "color": "#000000",
          "fontFamily": "AvenirNext-Medium",
          "fontSize": 22,
          "height": 70,
          "marginRight": 4,
          "textAlign": "center",
          "width": 60,
        },
        {},
        {
          "borderColor": "#333333",
        },
      ]
    }
    textAlign="center"
    value=""
  />
  <TextInput
    autoCorrect={false}
    autoFocus={false}
    keyboardType="numeric"
    maxLength={1}
    multiline={false}
    onChangeText={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    placeholder=""
    placeholderTextColor="#E4E4E4"
    style={
      [
        {
          "borderBottomWidth": 2,
          "borderColor": "#E9E9E9",
          "color": "#000000",
          "fontFamily": "AvenirNext-Medium",
          "fontSize": 22,
          "height": 70,
          "marginRight": 4,
          "textAlign": "center",
          "width": 60,
        },
        {},
        {
          "borderColor": "#333333",
        },
      ]
    }
    textAlign="center"
    value=""
  />
  <TextInput
    autoCorrect={false}
    autoFocus={false}
    keyboardType="numeric"
    maxLength={1}
    multiline={false}
    onChangeText={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    placeholder=""
    placeholderTextColor="#E4E4E4"
    style={
      [
        {
          "borderBottomWidth": 2,
          "borderColor": "#E9E9E9",
          "color": "#000000",
          "fontFamily": "AvenirNext-Medium",
          "fontSize": 22,
          "height": 70,
          "marginRight": 4,
          "textAlign": "center",
          "width": 60,
        },
        {},
        {
          "borderColor": "#333333",
        },
      ]
    }
    textAlign="center"
    value=""
  />
</View>
`;
